src/main/scala/edu/ie3/simona/scheduler/ScheduleLock.scala

```diff
@@ -196,34 +177,26 @@ object ScheduleLock {
     *   The scheduler to lock
     * @param awaitedKeys
     *   The keys that have to be supplied in order to unlock this lock
+    * @param expectedTick
+    *   The tick that an activation is expected for
     */
   private def apply(
       scheduler: ActorRef[SchedulerMessage],
       awaitedKeys: Set[UUID],
-  ): Behavior[LockMsg] =
-    Behaviors.withStash(100) { buffer =>
-      Behaviors.receiveMessage {
-        case Init(adapter) =>
-          buffer.unstashAll(uninitialized(scheduler, awaitedKeys, adapter))
-
-        case msg =>
-          // stash all messages until we are initialized
-          buffer.stash(msg)
-          Behaviors.same
-      }
-    }
-
-  private def uninitialized(
-      scheduler: ActorRef[SchedulerMessage],
-      awaitedKeys: Set[UUID],
-      adapter: ActorRef[Activation],
-  ): Behavior[LockMsg] =
+      expectedTick: Long,
+  ): Behavior[Message] =
     Behaviors.withStash(100) { buffer =>
-      Behaviors.receiveMessagePartial {
-        case LockActivation =>
-          buffer.unstashAll(active(scheduler, awaitedKeys, adapter))
-
-        case unlock: Unlock =>
+      Behaviors.receivePartial {
+        case (ctx, Activation(tick)) =>
+          if (tick == expectedTick)
+            buffer.unstashAll(active(scheduler, awaitedKeys))
+          else
+            ctx.log.error(
+              s"Received lock activation for tick $tick, but expected $expectedTick"
+            )
+            Behaviors.stopped
```

Comment:
Why did you change this to the new syntax?

Reply:
Why not?